protocol/driver.go

```diff
@@ -190,3 +228,108 @@ func (m *ParseResponse) Nodes() (nodes.Node, error) {
 	}
 	return ast, err
 }
+
+// Version implements DriverHostClient.
+func (c *client) Version(rctx context.Context) (driver.Version, error) {
+	sp, ctx := opentracing.StartSpanFromContext(rctx, "bblfsh.client.Version")
+	defer sp.Finish()
+
+	resp, err := c.h.ServerVersion(ctx, &VersionRequest{})
+	if err != nil {
+		return driver.Version{}, err
+	}
+	vers := (*driver.Version)(resp.Version)
+	return *vers, nil
+}
+
+// Languages implements DriverHostClient.
+func (c *client) Languages(rctx context.Context) ([]manifest.Manifest, error) {
+	sp, ctx := opentracing.StartSpanFromContext(rctx, "bblfsh.client.Languages")
+	defer sp.Finish()
+
+	resp, err := c.h.SupportedLanguages(ctx, &SupportedLanguagesRequest{})
+	if err != nil {
+		return nil, err
+	}
+	out := make([]manifest.Manifest, 0, len(resp.Languages))
+	for _, m := range resp.Languages {
+		dm := m.ToNative()
+		out = append(out, *dm)
+	}
+	return out, nil
+}
+
+// NewManifest converts driver manifest to the corresponding protocol message.
+func NewManifest(m *manifest.Manifest) *Manifest {
+	dm := &Manifest{
+		Name:     m.Name,
+		Language: m.Language,
+		Aliases:  m.Aliases,
+		Features: make([]string, 0, len(m.Features)),
+	}
+	if m.Version != "" || !m.Build.IsZero() {
+		dm.Version = &Version{
+			Version: m.Version,
+			Build:   m.Build,
+		}
+	}
+	switch m.Status {
+	case manifest.Inactive:
+		dm.Status = DevelopmentStatus_Inactive
+	case manifest.Planning:
+		dm.Status = DevelopmentStatus_Planning
+	case manifest.PreAlpha:
+		dm.Status = DevelopmentStatus_PreAlpha
+	case manifest.Alpha:
+		dm.Status = DevelopmentStatus_Alpha
+	case manifest.Beta:
+		dm.Status = DevelopmentStatus_Beta
+	case manifest.Stable:
+		dm.Status = DevelopmentStatus_Stable
+	case manifest.Mature:
+		dm.Status = DevelopmentStatus_Mature
+	default:
+		st, _ := strconv.Atoi(string(m.Status))
```

Comment:
I think if I read the code correctly, we could get the same effect by writing

```go
dm.Status = DevelopmentStatus(m.Status.Rank())
```

in this case, right?

Reply:
Hm, actually I think I see what's going on here: This case is to balance the other direction, where an unknown status code gets rendered as a text label.

Reply:
Yep, you are right with the second one.

Although this point and the one below reminds me that we could synchronize the rank with the enum on the protobuf side.

There are few issues though. The first one is that the rank map is private, so we can't do a reverse lookup and we don't want to export the map. The second is that if we decide to add an intermediate status/rank it will go out of sync with protobuf (can be solved with a comment on the rank map).

Reply:
We can probably consider that (synchronization) for a future change. One thing that may help is that the protobuf generated code exports maps from strings to int values, and vice versa, for each `enum` in the .proto file. We could either import that and use it to populate the rank map automatically (in `init`) or use it to generate the map on demand with a `go generate` rule.